Editor/Utility/SaveDuringPlay.cs

```diff
@@ -385,28 +383,19 @@ static object LeafObjectFromString(Type type, string value, GameObject[] roots)
 
         static string StringFromLeafObject(object obj)
         {
-            if (obj == null)
-                return string.Empty;
-
-            if (typeof(Component).IsAssignableFrom(obj.GetType()))
-            {
-                Component c = (Component)obj;
-                if (c == null) // Component overrides the == operator, so we have to check
-                    return string.Empty;
-                return ObjectTreeUtil.GetFullName(c.gameObject);
-            }
-            if (typeof(GameObject).IsAssignableFrom(obj.GetType()))
+            switch (obj)
             {
-                GameObject go = (GameObject)obj;
-                if (go == null) // GameObject overrides the == operator, so we have to check
+                case null:
                     return string.Empty;
-                return ObjectTreeUtil.GetFullName(go);
-            }
-            if (typeof(ScriptableObject).IsAssignableFrom(obj.GetType()))
-            {
-                return AssetDatabase.GetAssetPath(obj as ScriptableObject);
+                case Component c:
+                    return c == null ? string.Empty : ObjectTreeUtil.GetFullName(c.gameObject);
+                case GameObject go:
+                    return ObjectTreeUtil.GetFullName(go);
+                case ScriptableObject so:
+                    return AssetDatabase.GetAssetPath(so);
+                default:
+                    return obj.ToString();
             }
-            return obj.ToString();
         }
```

Comment:
I'm uneasy about this change.  I'm wondering if it properly handles Unity's overriding of the == null operator (which will return true for non-null objects that have been deleted).  I'm doubtful that the c# hocus-pocus here will handle those cases the same way.

Reply:
Ok. I checked and it looked like it handles "null" cases too, but to be sure I reverted it back to the original state.